client/src/commands/run.ts

```diff
@@ -52,7 +62,13 @@ function getCode(selected = false, uri?: Uri): string {
   } else {
     code = doc?.getText();
   }
+  if (codeFile) {
+    // Environment Variable SAS_EXECFILEPATH is set by SAS Enhanced Editor
+    // code = "option set=SAS_EXECFILEPATH=%sysfunc(quote(" + codeFile + "));\n" + code;
 
+    // Macro-variable &_SASPROGRAMFILE is set in SAS Studio and SAS Enterprise Guide
+    code = "%let _SASPROGRAMFILE = %bquote(" + codeFile + ");\n" + code;
```

Comment:
I think we need to make sure to not resolve any macro from the name. `%` and `&` are valid in file names.
I looked at SAS Studio and it writes `%nrquote(%nrstr());`, so that it doesn't resove macro in compilation and execution phases.
And it has to escape unmatched parentheses. To pass 2 macro quoting, it may need to be something like
`codeFile.replaceAll('(', '%%%(').replaceAll(')', '%%%)');`
And it would be good to extract the logic to a single place, like a function in `SasCodeHelper.ts`.

Reply:
Sorry I'm not very familiar with SAS Studio, how can you see/test that 
> it writes %nrquote(%nrstr());

?
I see the point of writing a function in `SasCodeHelper.ts`.
Any suggestion on how to organize unit tests for such a function ?

Reply:
> how can you see/test that

It's submitted SAS code so can be seen in SAS log.
So the overall code will be something like

```suggestion
    codeFile = codeFile.replaceAll('(', '%%%(').replaceAll(')', '%%%)');
    code = "%let _SASPROGRAMFILE =%nrquote(%nrstr(" + codeFile + ");\n" + code;
```

> Any suggestion on how to organize unit tests for such a function ?

Unit test can be in corresponding `test` directory, that is `client/test/components/Helper/SasCodeHelper.test.ts`

Reply:
The `replaceAll` method is part of ES2021 but tsconfig.json specifies "lib": ["ES2019", "DOM"],
how should that be solved ?

Reply:
Oh, didn't notice that. It can be achieved by
`codeFile = codeFile.replace(/[()]/g, '%%%$&');`

Reply:
I also added quoting for unmatched quotes.
I found your suggestion was generating extra % signs.
So I ended up with: `codeFile = codeFile.replace(/[('")]/g, "%$&");`